src/NUnitFramework/testdata/TestCaseAttributeFixture.cs

```diff
@@ -114,5 +114,10 @@ public void MethodWithExcludeRuntime(int num)
         public void MethodWithArrayArguments(object o)
         {
         }
+
+        [TestCase("doesn't work", TypeArgs = new[] { typeof(int) })]
```

Comment:
I feel an Analyzer rule request coming up.

Reply:
True, analyzers are a great validator for what would otherwise be runtime checks. What do you think about having a generic attribute to extend this on modern (NET8+) runtimes after we get the framework targeting that TFM?

Reply:
Do we really need to wait for .NET8? It is a C#11 feature and it seems to work in .NET6.0 but fails in .NET Framework. with a runtime error: `Generic types are not valid`

```csharp
    /// <summary>
    /// Marks a method as a parameterized test suite and provides arguments for each test case.
    /// </summary>
    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
    public class TestCaseAttribute<T> : TestCaseAttribute
    {
        /// <summary>
        /// Construct a TestCaseAttribute with a list of arguments.
        /// </summary>
        public TestCaseAttribute(T argument)
            : base(new object?[] { argument })
        {
            TypeArgs = new[] { typeof(T) };
        }
    }
```

And equivalent version with 2, 3, 4, etc. parameters.

With a usage :

```
        [TestCase(2, TypeArgs = new[] { typeof(double) }, ExpectedResult = typeof(double))]
        [TestCase<double>(2, ExpectedResult = typeof(double))]
        public Type GenericMethodAndParameterWithExplicitOrImplicitTyping<T>(T input)
            => typeof(T);
```